urldee901fae6_DocX_personal_timetabling_tgz-pJ8-net_personaltt_main_BenchmarkJ8/src/net/personaltt/solver/heuristics/BestAllocationSelection.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package net.personaltt.solver.heuristics;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Random;
+import net.personaltt.model.Occurrence;
+import net.personaltt.model.OccurrenceAllocation;
+import net.personaltt.solver.core.AllocationSelection;
+import net.personaltt.solver.core.SolverState;
+import net.personaltt.utils.BaseInterval;
+import net.personaltt.utils.RandomUtils;
+import net.sf.cpsolver.ifs.util.DataProperties;
+
+/**
+ * Best allocation selection. Allocation is better if has smaller cost, otherwise if is longer.
+ * Selection values allocations on conflicts count basis, thus it can use
+ * iteration over aligned allocations to elementary intervals in schedule.
+ * 
+ * @author docx
+ */
+public class BestAllocationSelection implements AllocationSelection {
+
+    Random random = new Random();
+    
+    double conflictCostWeight;
+    double valueCostWeight;
+
+    public BestAllocationSelection(DataProperties properties) {
+        conflictCostWeight = properties.getPropertyDouble("bestAllocationSelection.conflictCostWeight", 1.0);
+        valueCostWeight = properties.getPropertyDouble("bestAllocationSelection.valueCostWeight", 1.0);
+    }
+
+    
+    @Override
+    public OccurrenceAllocation select(SolverState schedule, Occurrence forOccurrence) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.